test/util/GTestHelpers.h

```diff
@@ -124,3 +124,27 @@ MATCHER_P(InsertIntoStream, matcher,
   *result_listener << "that yields \"" << output << "\"";
   return testing::ExplainMatchResult(matcher, output, result_listener);
 }
+
+// Helper type that allows to use non-copyable types in gtest matchers.
+template <typename T>
+class CopyShield {
+  std::shared_ptr<T> pointer_;
```

Comment:
I would argue for a `variant<shared_ptr<T>, reference_wrapper<const T>` here,
Which uses the `reference_wrapper` for the case, that a non-copyable type is passed in via a
`const &`, 
s.t. I can write
``` 
IdTable expected = ...
EXPECT_THAT(something, Eq(CopyShield(expected));
```
Your current implementation would require a `expected.clone()` here and doesn't work for truly uncopyable types in this scenario.

Reply:
In this scenario we have two constructors:

One from `const T&` (stores into the reference_wrapper)
One variadic one, that requires cosntructibility of T with the arguments and uses the shared_ptr.

Reply:
In this case you could safely use `std::cref` though.